src/main/java/at/hannibal2/skyhanni/data/BitsAPI.kt

```diff
@@ -0,0 +1,147 @@
+package at.hannibal2.skyhanni.data
+
+import at.hannibal2.skyhanni.events.ConfigLoadEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.ProfileJoinEvent
+import at.hannibal2.skyhanni.events.ScoreboardChangeEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.removeResets
+import at.hannibal2.skyhanni.utils.StringUtils.trimWhiteSpaceAndResets
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object BitsAPI {
+    var bits = 0
+    var currentFameRank = FameRank.NEW_PLAYER
+    var bitsToClaim = 0
+
+    private const val defaultcookiebits = 4800
+
+    private val group = RepoPattern.group("data.bits")
+    val bitsScoreboardPattern by group.pattern(
+        "scoreboard.bits",
+        "^Bits: §b(?<amount>(,?\\d{1,3})*)(.\\d+)?( ?(§?3?(\\((?<earned>[+-](,?\\d)*)?\\))?)?)?$"
+    )
+    private val bitsFromFameRankUpChatPattern by group.pattern(
+        "chat.bitsfromfamerankup",
+        "§eYou gained §3(?<amount>.*) Bits Available §ecompounded from all your §epreviously eaten §6cookies§e! Click here to open §6cookie menu§e!"
+    )
+    private val bitsEarnedChatPattern by group.pattern("chat.earned", "§f\\s+§8\\+§b(?<amount>.*)\\s+Bits\n")
+    private val boosterCookieAte by group.pattern("chat.boostercookieate", "§eYou consumed a §6Booster Cookie§e! §d.*")
+    private val bitsAvailableMenu by group.pattern(
+        "gui.bitsavailablemenu",
+        "§7Bits Available: §b(?<toClaim>[\\w,]+)(§3.+)?"
+    )
+    private val fameRankSbmenu by group.pattern("gui.fameranksbmenu", "§7Your rank: §e(?<rank>.*)")
+    private val fameRankCommunityShop by group.pattern("gui.famerankcommunityshop", "§7Fame Rank: §e(?<rank>.*)")
+
+    @SubscribeEvent
+    fun onScoreboardChange(event: ScoreboardChangeEvent) {
+        for (line in event.newList) {
+            val message = line.trimWhiteSpaceAndResets().removeResets()
```

Comment:
no need to double remove resets
```suggestion
            val message = line.trimWhiteSpace().removeResets()
```